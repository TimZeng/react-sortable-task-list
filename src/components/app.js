import React, { Component } from 'react';

import SortableListComponent from './sortableList';

export default class App extends Component {
  render() {
    return (
      <div className='container'>
        <SortableListComponent />
      </div>
    );
  }
};