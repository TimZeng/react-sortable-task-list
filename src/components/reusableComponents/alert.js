import React from 'react';

export const Alert = ({ alert, onClose }) => {
  if ( !alert.message ) {
    return null;
  }

  return (
    <div
      className='alert'
      style={{
        ...styles.alertStyle,
        ...styles[`${alert.status===200 ? 'successStyle' : 'errorStyle'}`]
      }}>
      <span>{ alert.message }</span>
      <i style={{lineHeight: '21px'}} onClick={onClose} className="button fa fa-times" />
    </div>
  )
};

const styles = {
  alertStyle: {
    position: 'fixed',
    bottom: '25%',
    right: '5%',
    width: '500px',
    maxWidth: '80%',
    display: 'flex',
    justifyContent: 'space-between',
    padding: '8px 15px',
    fontSize: '70%',
    borderRadius: '3px',
    zIndex: '2'
  },

  successStyle: {
    backgroundColor: '#f7fffb',
    color: '#84d9a7',
    border: '1px solid #84d9a7'
  },

  errorStyle: {
    backgroundColor: '#fff7f7',
    color: '#d98484',
    border: '1px solid #d98484'
  }
}