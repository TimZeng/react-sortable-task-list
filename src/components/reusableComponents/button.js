import React from 'react';

export const Button = ({ text, divStyle = {}, onClick, disabled = false }) => (
  <div
    className={ `button ${disabled?'disabled':''}` }
    style={{ ...styles.buttonStyle, ...divStyle }}
    onClick={ disabled ? null : onClick }
  >
    { text }
  </div>
);

const styles = {
  buttonStyle: {
    display:'inline-block',
    padding: '8px 15px',
    borderRadius: '5px'
  }
};