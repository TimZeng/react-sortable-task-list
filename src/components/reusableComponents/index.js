export * from './button';
export * from './inputGroup';