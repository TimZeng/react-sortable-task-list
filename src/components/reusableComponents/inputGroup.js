import React from 'react';

export const InputGroup = ({ actionText = 'Submit', value, onChange, onClick }) => (
  <div
    className='input-group'
    style={ styles.inputGroupStyle }
  >
    <input
      style={ styles.inputStyle }
      value={value}
      onChange={e => onChange(e.target.value)}
      placeholder='Enter task name'
      autoFocus
    />
    <div
      className={`button ${value==='' ? 'disabled' : ''}`}
      style={styles.buttonStyle}
      onClick={ value==='' ? null : onClick }>{actionText}
    </div>
  </div>
);

const styles = {
  inputGroupStyle: {
    backgroundColor: '#fff',
    justifyContent: 'space-between',
    padding: '10px',
    marginBottom: '15px',
    border: '2px solid #78da9f'
  },

  inputStyle: {
    width: '90%'
  },

  buttonStyle: {
    height: '34px',
    backgroundColor: '#78da9f',
    padding: '5px 10px',
    lineHeight: '25px',
    textAlign: 'center',
    borderRadius: '3px',
    color: '#fff'
  }
}