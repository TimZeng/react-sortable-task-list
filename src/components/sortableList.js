import React, { Component } from 'react';
import { SortableContainer, SortableElement } from 'react-sortable-hoc';

const SortableItem = SortableElement(({itemIdex, value, onDelete}) => {
  return (
    <div className="light-text" style={styles.listItemStyle}>
      <span>
        <i className="fa fa-tasks" />
        &nbsp;&nbsp;&nbsp;&nbsp;
        {value}
      </span>
      <i
        onClick={() => onDelete(itemIdex)}
        style={{lineHeight: '30px'}}
        className="button fa fa-trash-o"
      />
    </div>
  );
});

const SortableList = SortableContainer(({items, onDelete}) => {

  return (
    <div>
      {items.map((value, index) => (
        <SortableItem key={`item-${index}`} index={index} itemIdex={index} value={value} onDelete={onDelete} />
      ))}
    </div>
  );
});

export default class SortableListComponent extends Component {
  shouldCancelStart(e) {
    if (e.target.tagName.toLowerCase() === 'i') {
      return true;
    }
  }

  render() {
    return (
      <div className='sortable-list'>
        <SortableList
          items={this.props.items}
          onSortEnd={this.props.onSortEnd}
          onDelete={this.props.onDelete}
          shouldCancelStart={this.shouldCancelStart}
        />
      </div>
    );
  }
};

const styles = {
  listItemStyle: {
    backgroundColor: '#fff',
    padding: '10px',
    marginBottom: '15px',
    boxShadow: '0px 2px 9px -1px rgba(145,140,145,0.3)',
    borderRadius: '3px',
    // minHeight: '200px',
    display: 'flex',
    justifyContent: 'space-between'
  }
}