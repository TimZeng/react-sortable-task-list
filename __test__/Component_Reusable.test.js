import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-15';
Enzyme.configure({ adapter: new EnzymeAdapter() });
import toJson from 'enzyme-to-json';

import { Button, Alert } from '../src/components/reusableComponents';

describe('Button component should render dynamically', () => {
  const text = 'button';
  const disabled = true;

  it('should render Button as expected', () => {
    const props = {text};
    const component = shallow(<Button {...props} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
    expect(component.find('div.button').length).toBe(1);
    expect(component.find('div.disabled').length).toBe(0);
    expect(component.find('div.button').text()).toEqual(text);
  })

  it('should disable Button', () => {
    const props = {text, disabled};
    const component = shallow(<Button {...props} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
    expect(component.find('div.button.disabled').length).toBe(1);
    expect(component.find('div').props().onClick).toEqual(null);
  })

});

describe('Alert component should render dynamically', () => {
  const successAlert = { status: 200, message: 'success' };
  const errorAlert = { status: 400, message: 'error' };

  it('should render Success Alert as expected', () => {
    const props = {alert: successAlert};
    const component = shallow(<Alert {...props} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
    expect(component.find('span').text()).toEqual(successAlert.message);
  })

  it('should render Error Alert as expected', () => {
    const props = {alert: errorAlert};
    const component = shallow(<Alert {...props} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
    expect(component.find('span').text()).toEqual(errorAlert.message);
  })

  it('should NOT render Alert', () => {
    const props = {alert: {}};
    const component = shallow(<Alert {...props} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
    expect(component.find('div.alert').length).toBe(0);
  })

});
