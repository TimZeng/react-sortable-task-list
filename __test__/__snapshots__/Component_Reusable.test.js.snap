// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component should render dynamically should NOT render Alert 1`] = `""`;

exports[`Alert component should render dynamically should render Error Alert as expected 1`] = `
<div
  className="alert"
  style={
    Object {
      "backgroundColor": "#fff7f7",
      "border": "1px solid #d98484",
      "borderRadius": "3px",
      "bottom": "25%",
      "color": "#d98484",
      "display": "flex",
      "fontSize": "70%",
      "justifyContent": "space-between",
      "maxWidth": "80%",
      "padding": "8px 15px",
      "position": "fixed",
      "right": "5%",
      "width": "500px",
      "zIndex": "2",
    }
  }
>
  <span>
    error
  </span>
  <i
    className="button fa fa-times"
    style={
      Object {
        "lineHeight": "21px",
      }
    }
  />
</div>
`;

exports[`Alert component should render dynamically should render Success Alert as expected 1`] = `
<div
  className="alert"
  style={
    Object {
      "backgroundColor": "#f7fffb",
      "border": "1px solid #84d9a7",
      "borderRadius": "3px",
      "bottom": "25%",
      "color": "#84d9a7",
      "display": "flex",
      "fontSize": "70%",
      "justifyContent": "space-between",
      "maxWidth": "80%",
      "padding": "8px 15px",
      "position": "fixed",
      "right": "5%",
      "width": "500px",
      "zIndex": "2",
    }
  }
>
  <span>
    success
  </span>
  <i
    className="button fa fa-times"
    style={
      Object {
        "lineHeight": "21px",
      }
    }
  />
</div>
`;

exports[`Button component should render dynamically should disable Button 1`] = `
<div
  className="button disabled"
  onClick={null}
  style={
    Object {
      "borderRadius": "5px",
      "display": "inline-block",
      "padding": "8px 15px",
    }
  }
>
  button
</div>
`;

exports[`Button component should render dynamically should render Button as expected 1`] = `
<div
  className="button "
  style={
    Object {
      "borderRadius": "5px",
      "display": "inline-block",
      "padding": "8px 15px",
    }
  }
>
  button
</div>
`;
